scikeras/_utils.py

```diff
@@ -133,8 +137,6 @@ def make_model_picklable(model_obj):
     Model
         The input model, but directly picklable.
     """
-    if not isinstance(model_obj, Model):
-        raise TypeError("`model_obj` must be an instance of a Keras Model")
     model_obj.__reduce_ex__ = pack_keras_model.__get__(model_obj)
```

Comment:
I think we can hold off on any of these types of changes to see where the TF Pickle RFC goes.

I only deleted that line above because we were really close to 100% coverage and I just wanted to see everything green without implementing a useless test. I'm only human, I got greedy.

Reply:
I'd rather see a test implemented to get to 100% test coverage rather code. The behavior the code implements is far more important than the tests.

Is this assertion relevant? `make_model_picklable` is only called two places in `wrappers.py`, both times with a Keras model. Either way, the test is trivial:

``` python
def test_make_model_picklable_input():
    with pytest.raises(ValueError, match="instance of Keras Model"):
        make_model_picklable("not even a model")
```

Reply:
I agree with you. To be clear, my thought process was indeed "we only ever use this in one or two places internally so do we really need this check? If we want to keep this check then we should implement a test for it. So do we really want to keep the check and the test for a bug that will probably never happen?" And my reply to this internal monologue was "no, we should just delete the check".

Reply:
I'm going to remove this change from this PR since it proved to be controversial (with reason). I'll make a separate PR for it since I still think we should remove this assertion, as suggested above.

Reply:
:+1:  Python type checks might help with documentation too.